File: plugins/ffmpeg-transcoder/src/ViewItPlugin.java
package ai.viewit.app;

import android.content.Context;
import android.net.Uri;
import java.io.File;
import java.util.List;

public interface ViewItPlugin {
    String getId();
    String getVersion();
    List<String> getSupportedFormats();
    void initialize(Context context);
    boolean canHandle(String mimeType);
    boolean canHandleExt(String extension);
    boolean transcode(Uri inputUri, File outputFile, PluginProgress onProgress);
    void cleanup();
}

File: apps/mobile/src-tauri/android/ViewItPlugin.kt
package ai.viewit.app

import android.content.Context
import android.net.Uri
import java.io.File

interface ViewItPlugin {
    val id: String
    val version: String
    val supportedFormats: List<String>

    fun initialize(context: Context)
    fun canHandle(mimeType: String): Boolean
    fun canHandleExt(ext: String): Boolean
    fun transcode(input: Uri, output: File, onProgress: PluginProgress): Boolean
    fun cleanup()
}
